main.go

```diff
@@ -28,6 +28,8 @@ const (
 
 var version = "<not set>"
 
+var stayOnUntil = time.Now()
```

Comment:
The dbus handler will likely run in another goroutine so you need to guard `stayOnUntil` with a `sync.Mutex` to avoid data races. The mutex needs to locked whenever it is read and written.